File: WindowsForm_sample/UITest/ScenarioTest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;
using System.Threading.Tasks;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace UITestSample.Test
{
    [TestClass]
    public class ScenarioTest : FormSession
    {
        [TestMethod]
        public void ボタンを押すとHelloWorlが表示される()
        {
            Thread.Sleep(TimeSpan.FromSeconds(2));
            session.FindElementByName("button1").Click();
            string result = session.FindElementByAccessibilityId("textBox1").Text;
            Assert.AreEqual("Hello World", result);
        }

        [ClassInitialize]
        public static void ClassInitialize(TestContext context)
        {
            Setup(context);
        }

        [ClassCleanup]
        public static void ClassCleanup()
        {
            TearDown();
        }

    }
}

File: VBMSTest/ScenarioTest.vb
Imports System
Imports System.Threading
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports OpenQA.Selenium.Appium.Windows
Imports Xunit

Namespace VBMSTest
    'MSTestのアノテーションだがこれないとテストが実行されない（どういうことだよ.....?ｗ）
    <TestClass>
    Public Class ScenarioTest
        Inherits FormSession

        'MSTestのアノテーションで使えないのでコメントアウト
        '<ClassInitialize>
        Public Shared Sub ClassInitialize()
            Setup()
        End Sub

        ''MSTestのアノテーションで使えないのでコメントアウト
        '<ClassCleanup>
        Public Shared Sub ClassCleanup()
            TearDown()
        End Sub

        ' XUnitのアノテーションで使えるのでこちらを使用
        <Fact>
        Public Sub ボタンを押すとHelloWorldが表示される()
            Setup()
            Thread.Sleep(TimeSpan.FromSeconds(2))
            session.FindElementByName("Button1").Click()
            Dim result As String = session.FindElementByAccessibilityId("TextBox1").Text
            Microsoft.VisualStudio.TestTools.UnitTesting.Assert.AreEqual("HelloWorld!", result)
            TearDown()
        End Sub
    End Class

End Namespace
